packages/graphql-auth-transformer/src/resources.ts

```diff
@@ -162,8 +171,35 @@ export class ResourceFactory {
                 ])
             ])
         ])
+
+        const customGroupClaim: Expression = block('Static Group Authorization Checks with Custom Groups', [
```

Comment:
There is a lot of code duplication here that could be avoided. You are doing three extra steps: adding a comment when a custom claim is used, calling a different method to set userGroups, and handling the claim field being either a string or an array. You could remove this duplication (and source of future mis-alignment / bugs) by updating setCustomClaim function to convert any string claim to an array. Further, if you change setUserGroups to take a claim name as the first parameter with a default value of 'cognito:groups' you can completely remove this method, and remove the setCustomClaim method.

Reply:
Good call out - I've edited setUserGroups to allow an optional param for a custom group claim.